implementation/src/main/java/io/smallrye/graphql/execution/LRUCache.java

```diff
@@ -0,0 +1,100 @@
+package io.smallrye.graphql.execution;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Function;
+
+public class LRUCache<K, V> {
+
+    private final int maxSize;
+    private final Map<K, Entry<V>> cache = new ConcurrentHashMap<>();
+    private Entry<V> start;
+    private Entry<V> end;
+
+    LRUCache(int maxSize) {
+        this.maxSize = maxSize;
+    }
+
+    V get(K key) {
+        Entry<V> entry = cache.computeIfPresent(key, this::moveEntryToStart);
+        return entry == null ? null : entry.value;
+    }
+
+    V put(K key, V value) {
```

Comment:
This method is not thread safe.  NPEs and other errors are possible.  It isn't used at all.  Should it just be removed?